app/views/visual_groups/_form.html.erb

```diff
@@ -0,0 +1,39 @@
+<%= form_with(model: [@visual_model, @visual_group], local: true,
+              id: "visual_group_form") do |form| %>
+  <% if visual_group.errors.any? %>
+    <div id="error_explanation">
+      <h2>
+	<%= pluralize(visual_group.errors.count, "error") %>
+	prohibited this visual_group from being saved:
+      </h2>
+
+      <ul>
+        <% visual_group.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div>
+    <div class="field">
+      <%= form.label :name %>
+      <%= form.text_field(:name, size: 40) %>
+      <%= :VISUAL_GROUP.t %>
+    </div>
+
+    <div class="field">
+      <%= form.label :description %><br>
+      <%= form.text_area(:description, cols: 60, rows: 10) %>
+    </div>
+
+    <div class="field">
+      <%= form.label :approved %>
+      <%= form.check_box :approved %>
+    </div>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
```

Comment:
I forgot if you still want this, but if you want Rails to disable the `submit` button on submit, you do have to pass a `data-disable-with` attribute to the button helper (you can use something like `data-disable-with: "Filtering... #{safe_spinner}"` to get alternate text with a "spinner".) 
`rails-ujs` will disable the button until the form is re-rendered. 

Invalid fields will keep the submit button disabled, though... so, to re-enable the submit button when an input changes, i have added a little snippet on #1160 after line 51 in `assets/javascripts/main.js`. Not sure if you need this here, but i imagine so. 

```javascript
 // Not a great solution, but ok for now.
 jQuery('form :input').on('change', function() {
   var disabled_buttons = $('[data-disable-with]');
   $(disabled_buttons).each(function() {
     $.rails.enableElement(this);
   })
 });
```

Reply:
That submit button is not the issue I was concerned about.  I've been struggling with the extra form in edit.html.erb for updating the set of image that the user is viewing.  It's mostly working now, but it's still possible to get into a state where the button magically becomes disable for no good reason, but I've changed the way the UI work enough that I think it will be rare.

Reply:
Tried this in the edit.html.erb, but still no joy.  I added it verbatim at the beginning of the `inject_javascript_at_end` block.  The bad behavior is, create a visual group (start at http://localhost:3000/visual_models and do the CRUD thing) with the same name as some mushroom in your local database (e.g., "Agaricus campestris").  Goto to the edit Visual Group page.  Click "Include" or "Exclude" under some image.  Then click "Update Filter".  It may work correct the first time, but if you do it again the "Update Filter" button it won't refresh and it gets disabled and remains disabled until you reload the page either in the usual way or by clicking the "Reload" button.  Once it has failed once it will continue to fail.

Reply:
I'll grab the latest code and take a look at this shortly.  I bet it's something really small.  Have you checked if there are any javascript errors in the browser error console?

Reply:
I suspect you're right.  I did look for JS errors and didn't see anything obvious.  My JS debugging skills have never been very good and they are very rusty right now.

Reply:
@mo-nathan -- Urk, I don't see nimmo's suggested code snippet anywhere.  Is it possible I don't have your latest commits?

Reply:
@mo-nathan -- Well, I played with it anyway.  I'm at commit 848d888f5c81818d5f36da7387c82926624cb58d, by the way.  And I confirm your behavior.  The first time I ask it to update the filter it works, but subsequent clicks do nothing.  I don't think this has anything to do with javascript.  I think it's the anchor.  Firefox (at least for me) doesn't seem to see fit to actually reload the page when you click on same URL if there's a "#anchor" in it.  I bet if you removed that it would work fine...  Yes, that's correct, I just tried it.

Reply:
Here's what I did, for the record:

--- a/app/views/visual_groups/edit.html.erb
+++ b/app/views/visual_groups/edit.html.erb
@@ -28,7 +28,7 @@
   <hr>
 
 <strong><%= :edit_visual_group_filter_options.t %></strong>
-<%= form_with(url: edit_visual_group_path(@visual_group, anchor: "filter_options"), method: :get,
+<%= form_with(url: edit_visual_group_path(@visual_group), method: :get,
               id: "visual_group_filters_form") do |form| %>
   <%= form.text_field(:filter, value: @filter, size: 60, id: "filter") %> <%= form.submit(:edit_visual_group_update_filter.t) %>

Reply:
Nice find!  I played around a bit more and ended up going with your suggestion.  Dumped the JS I had in there since it wasn't actually being used on the page any more.

I didn't commit Nimmo's code.  It was earlier in this thread if you want to see it.

Reply:
That sounds right. Sorry for the red herrings.